index.js

```diff
@@ -392,17 +379,12 @@ let Fixtures = {
      * @return {Promise} The ChakramResponse for the delete request to the server.
      */
     cleanup() {
-        if (_.isUndefined(this._headers['OAuth-Token'])) {
-            if (++this._sessionAttempt > this._maxSessionAttempts) {
-                throw new Error('Max number of login attempts exceeded!');
-            }
-
-            return this._adminLogin().then(() => this.cleanup());
+        if (this._needsLogin()) {
```

Comment:
Also, note that for whatever reason we don't do this same check on `Fixtures.link`. I don't remember whether that was a deliberate decision or not.